[#326] Add a YAML tab on the Broker details view

Msarawan opened the pull request:
Added a YAML tab on the Broker Details view to make it easier for users to access the Broker's details.

![Screenshot from 2024-12-05 16-43-16](https://github.com/user-attachments/assets/03b24839-ff17-40d5-8e72-669b19217273)

fixes: [#326](https://github.com/artemiscloud/activemq-artemis-self-provisioning-plugin/issues/326)

lavocatt commented on src/brokers/broker-details/components/yaml/Yaml.container.tsx:
```diff
@@ -0,0 +1,48 @@
+  };
+
+  return (
+    <Page>
```
This can't be a `<Page>`, because it's inside a `<Page>` already, it can only be a `<PageSection>`

lavocatt reviewed (changes requested):
This is all very good work, congrats. The only thing that we need to focus on now is to fix the problem with the vertical space the yaml editor is taking.

If you look at all the other views, the yaml editor is always taking the maximum vertical space possible without going overboard. In contrary, in the implementation you have now, the buttons appear under the page.

lavocatt committed:
[#7] proxy the jolokia api server

In order to get authenticated as a user using the jolokia api server, we
need to set up proxies. The console will take care of adding the
authorization bearer token on every requests for us.

This commit adds the configuration for running both in dev mode and
production mode.

lavocatt reviewed (commented):
Good work! It's working now, that's good. There are a few things to improve in the PR itself:

- We need to split the work a bit, there's a first commit that is about adding a YAML tab with all it needs (including removing the Page and PageSection in the broker details page, and this entry point only) and there's another commit that is removing the Page and PageSection everywhere else, this has to be with another github issue number or NO-ISSUE as it's unrelated to adding the YAML tab in the broker details page, even though we've learned that Page and PageSection are bad doing this work.
- We need to have more tests to make sure there's no regressions.

Msarawan commented on src/brokers/broker-details/BrokerDetails.container.tsx:
```diff
@@ -50,135 +48,158 @@ const AuthenticatedPageContent: FC<AuthenticatedPageContentPropType> = ({
+      )}
+      {activeTab === 'clients' && <ClientsContainer />}
+      {activeTab === 'pods' && <PodsContainer />}
+      {activeTab === 'yaml' && !loadingBrokerCr && (
+        <YamlContainer brokerCr={brokerCr} />
+      )}
+      {loadingBrokerCr && <Spinner size="md" />}
```
sure, i will add...

howardgao committed:
[#1] Migration update

Updated artemiscloud references to arkmq-org

lavocatt committed:
[#7] jolokia api server without login phase

The new jolokia api server doesn't require to login to access its
services but requires to have a bearer token passed onto every requests
alongside the actual jolokia endpoint that is the target of the request.

howardgao committed:
[#1] Migration update

Added package manager

Msarawan committed:
[#326] Add a YAML tab on the Broker details view

Added a YAML tab on the Broker Details view to make it easier
for users to access the Broker's details.

Msarawan committed:
[NO-ISSUE] Refactor: remove unnecessary Page and PageSection usage from the codebase

Simplified the codebase by eliminating unnecessary Page and PageSection components,
improving layout consistency and aligning with best practices.

Msarawan commented:
This PR is recreated on the new `arkmq-org` repo: [PR#4](https://github.com/arkmq-org/activemq-artemis-self-provisioning-plugin/pull/4)

howardgao committed:
[#1] Migration update

Update workflow to use bot token
Without the token the bot cannot push to remote repo

Commit:
Update version to 0.3.1

howardgao committed:
[NO-ISSUE] Revert "Update version to 0.3.1"

This reverts commit 08f650224cf898d9255bb08de38e1c8509d12c68.

howardgao committed:
[#1] Migration update

Fix update.yaml workflow to update versions in package.json and
charts/openshift-console-plugin/Chart.yaml

howardgao committed:
[#1] Migration update

Updated update.yaml to update missing versions in package.json

Commit:
Update version to 0.3.1

howardgao committed:
[#1] Migration update

- Revert "Update version to 0.3.1"
This reverts commit 1f6f1e4d332a8c6901368e2a7a6aaf324b5e3530.
- Fix update workflow error (missing , in sed operation)

lavocatt committed:
[NO-ISSUE] getting the routes must be namespaced

Otherwise the user developer can't list the brokers.

Commit:
Update version to 0.3.1

howardgao committed:
[#1] Migration update
Revert bad commit of arkmq-bot due to bad package.json

Revert "Update version to 0.3.1"

This reverts commit 9bf581ba740666ca7dec25e232f6d998f86a785b.

howardgao committed:
[#1] Migration update
Fix sed error: missing quote in expression

Commit:
Update version to 0.3.1

Commit:
Update version to 0.4.0
